Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -97,7 +97,19 @@
      * The decay factor used for generating the weight of the point. For greater
      * values of lambda we become more biased in favor of recent points.
      */
-    private final double lambda;
+    private double lambda;
+    /**
+     * The last timestamp when lambda was changed
+     */
+    private long lastUpdateOflambda = 0;
+    /**
+     * most recent timestamp
+     */
+    private long mostRecentTimeStamp = 0;
+    /**
+     * The accumulated sum of lambda before the last update
+     */
+    private float accumulatedLambda = 0;
```

Comment:
question. is expected to reset the variable to 0 for deserialized models?

Reply:
No - this is a bug! Fixing.